sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1727,7 +1727,7 @@ def test_sgd_oneclass():
     assert_allclose(clf.decision_function(X_test), dec)
 
     pred = clf.predict(X_test)
-    assert_array_equal(pred, np.array([-1, 1]))
+    assert_array_equal(pred, np.array([1, 1]))
```

Comment:
Why the new nu value and outputs ? If I'm not mistaken, the changes in this PR should give the same results as before  (in main we divide nu by 2 and then multiply by 2 in the sgd code, in this PR we simply keep nu as is) ?

Reply:
> `alpha` is used elsewhere too not just where we changed it within the Cython code.

Ah yes good point! AFAIK `alpha` is also used for the weights updates, and for the `learning rate='optimal'` option. I think the changes in this PR are actually a bug fix: the updates were incorrect in main (the weights and intercept updates were using different "alpha" values).

Do you think we can come up with a non regression test to check that we have now the correct updates and  the updates in main were wrong ? I feel it's tough because SGDOneClassSVM is stochastic. Maybe extending `test_ocsvm_vs_sgdocsvm` on more difficult cases ?